api/server.spec.js

```diff
@@ -0,0 +1,50 @@
+const supertest = require('supertest')
+const server = require('./server');
+const db = require('../database/dbConfig');
+
+beforeEach(async () => {
+  // this function executes and clears out the table before each test
+  await db('users').truncate();
+});
+
+test('welcome route', async () => {
+  const res = await supertest(server).get('/')
+    expect(res.status).toBe(200)
+    expect(res.type).toBe('application/json')
+    expect(res.body.message).toMatch(/Sprint Challenge Authentication/i)
+});
+
+test('add a user to the database', async () => {
+  const res = await supertest(server)
+    .post('/api/auth/register')
+    .send({
+      "username": "test",
+      "password": "test"
+    })
+  expect(res.status).toBe(201)
+  expect(res.type).toBe('application/json')
+  expect(res.body.username).toBeString()
+});
+
+test('login as a test user and return a token', async () => {
+  const res = await supertest(server)
+    .post('/api/auth/login')
+    .send({
+      "username": "test",
+      "password": "test"
+    })
+  expect(res.status).toBe(200)
+  expect(res.type).toBe('application/json')
+  expect(res.body.token).toBeTruthy()
+});
+
+test('GET jokes from the api', () => {
```

Comment:
this test probably doesn't work either because your token is static, but a token will expire and needs to be created when a user is logging in

Reply:
I would login and save that token and then use that variable to go to the jokes page